src/posix/platform/netif.cpp

```diff
@@ -434,6 +604,7 @@ void platformNetifInit(otInstance *aInstance, const char *aInterfaceName)
 #if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
     platformUdpInit(sTunName);
 #endif
+    mldListenerInit();
```

Comment:
Seems like we should also have a check for:

if (sMLDMonitorFd != -1)
{
     close(sMLDMonitorFd);
     sMLDMonitorFd = -1;
}

at the end of this routine, right?

Reply:
In `mldListenerInit`, we die on errors.

Reply:
Fair enough, though it seems like that violates the symmetry of the code.  Some routines succeed or jump to exit on error.  Some die.  Why the difference?  Why not make them all the same, and clean up after all of them on exit?

Reply:
The difference lies on whether it is critical for further code executions.
For instance, an error in parsing the MLD report won't affect the future packet parsing while not being able to setup the listener will prevent you from parsing any MLD packets.

Reply:
@rob-the-dude I found you are right when I was working on https://github.com/openthread/openthread/pull/4726.
Would you mind to wait until https://github.com/openthread/openthread/pull/4726 is merged so that we can elegantly handle deinitialization?

Reply:
Sure.  I think #4726 is a good solution to making this cleaner and easier to understand.  Thank you.